ReadASCIIApp/src/ReadASCII.cpp

```diff
@@ -499,28 +487,26 @@ int ReadASCII::getSPInd (double SP)
     }
     if (rowNum > 0)
     {
-        std::cerr << "ReadASCII: SP above Look Up Higher Range, " << SP << " > " << pSP_[rowNum - 1] << std::endl;
+        std::cerr << "ReadASCII: SP above Look Up Higher Range, " << SP << " > " << getSetpointValue(rowNum-1) << std::endl;
     }
     return rowNum - 1;
 }
 
-void ReadASCII::updatePID(int index)
+void ReadASCII::updateControlValues(int index)
 {
     if (index < 0 || index >= rowNum)
     {
         return;
     }
-    // Set to minus one first and then the actual value.
-    // This marks the value as "changed" so that the monitor actually gets fired.
-    setDoubleParam(P_P, -1);
-    setDoubleParam(P_P, pP_[index]);
-    setDoubleParam(P_I, -1);
-    setDoubleParam(P_I, pI_[index]);
-    setDoubleParam(P_D, -1);
-    setDoubleParam(P_D, pD_[index]);
-    setDoubleParam(P_MaxHeat, -1);
-    setDoubleParam(P_MaxHeat, pMaxHeat_[index]);
-    std::cerr << "ReadASCII: Updating P=" << pP_[index] << " I=" << pI_[index] << " D=" << pD_[index] << " MP=" << pMaxHeat_[index] << std::endl;
+    std::unordered_map<std::string, int>::iterator it;
+    std::string prnt = "READASCII: Setting new control and tuning values: ";
+    for (it = pv_index_values_updated.begin(); it != pv_index_values_updated.end(); it++)
+    {
+        setDoubleParam(it->second, -1);
+        epicsFloat64 ret = setParamTableValue(it->first, it->second, index);
+        prnt += it->first + ":" + std::to_string(static_cast<long double>(ret)) + ", ";  // long double cast needed for VS2010
+    }
+    std::cerr << prnt + "\n";
```

Comment:
Using an `ostringstream` rather than string concatenation might be better - write to this as is you were writing to `cerr` and at the end use   `std::cerr << oss.str() << std::endl`